axum-debug/src/lib.rs

```diff
@@ -170,11 +189,31 @@ mod debug_handler {
         Ok(tokens)
     }
 
-    struct Attrs;
+    struct Attrs {
+        body_ty: Type,
+    }
 
     impl Parse for Attrs {
-        fn parse(_input: syn::parse::ParseStream) -> syn::Result<Self> {
-            Ok(Self)
+        fn parse(input: syn::parse::ParseStream) -> syn::Result<Self> {
+            let mut body_ty = None;
+
+            while !input.is_empty() {
+                let ident = input.parse::<syn::Ident>()?;
+                if ident == "body" {
+                    input.parse::<Token![=]>()?;
+                    body_ty = Some(input.parse()?);
+                } else {
+                    return Err(syn::Error::new_spanned(ident, "unknown argument"));
+                }
+
+                let _ = input.parse::<Token![,]>();
```

Comment:
I think the idiomatic way to do this is [Punctuated::parse_terminated_with](https://docs.rs/syn/latest/syn/punctuated/struct.Punctuated.html#method.parse_terminated_with)

Reply:
Hm how would that work 🤔? I'm not parsing multiple of the same thing. Specifically just parsing `body = Type` with optional trailing comma.

Reply:
The way I usually do this kind of thing is

```rust
enum MyAttribute {
    Body(syn::Type),
    // other variants may be added in the future
}

impl Parse for MyAttribute { /* ... */ }

let attributes = input.call(Punctuated::<MyAttribute, Token![,]>::parse_terminated)?;

let mut body = None;
for attr in attributes {
    match attr {
        // Or with a function that raises a duplicate-attribute
        // `syn::Error` if the `Option` is already `Some`
        MyAttribute::Body(b) => body = Some(b),
    }
}
```

Reply:
How do you handle having multiple variants in `MyAttribute` then, given that you end up with a vec of enums and would have to search through it to find a `MyAttribute::Body` to get the value of that attribute. Maybe I'm misunderstanding you but feels a struct with fields is easier to access.

Reply:
That's what the for loop is for, same as the while loop in your code. It's just that it's mostly decoupled from the parsing. The `body` var in my example above was intended to mirror `body_ty` in the code.

It's probably more code than what you wrote actually, but more declarative, less imperative.

Reply:
Ah okay yeah I get it now. I think I'll just leave it as it is. If it gets annoying if/when we add more attributes we can always reconsider.